sdk/src/client/node_manager/mod.rs

```diff
@@ -214,11 +216,8 @@ impl NodeManager {
                                 result_counter += 1;
                             },
                         ),
-                        Err(Error::ResponseError { code: 404, url, .. }) => {
-                            error.replace(crate::client::Error::NotFound(url));
-                        }
                         Err(err) => {
-                            error.replace(err);
+                            error.replace(err.into());
```

Comment:
Maybe a list of errors would be better here?

Reply:
Or a map of node id to error

Reply:
i agree, but thats quite out of scope of the issue i think. Wed still only return 1 error though so which one would you pick?

Reply:
Well, probably it would make sense to have an error type that has multiple inner errors.